File: server/datasets/python_source-code/Multiple-E3/type2_prefix_sum_clone_1_max_subarray.py
def largest_subarray(array_vals):
    best_total = float('-inf')
    running_prefix = 0
    lowest_prefix = 0
    
    for value in array_vals:
        running_prefix += value
        if running_prefix - lowest_prefix > best_total:
            best_total = running_prefix - lowest_prefix
        if running_prefix < lowest_prefix:
            lowest_prefix = running_prefix
            
    return best_total

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_e3/type2_prefix_sum_clone_1_max_subarray.java
public class Solution {
    public int largest_subarray(int[] array_vals) {
        int best_total = Integer.MIN_VALUE;
        int running_prefix = 0;
        int lowest_prefix = 0;
        
        for (int value : array_vals) {
            running_prefix += value;
            if (running_prefix - lowest_prefix > best_total) {
                best_total = running_prefix - lowest_prefix;
            }
            if (running_prefix < lowest_prefix) {
                lowest_prefix = running_prefix;
            }
        }
        return best_total;
    }
}
